libportal/settings.c

```diff
@@ -0,0 +1,234 @@
+      return;
+    }
+
+  g_variant_get_child (parameters, 0, "&s", &namespace);
+  g_variant_get_child (parameters, 1, "&s", &key);
+  g_variant_get_child (parameters, 2, "v", &value);
+
+  g_signal_emit (settings, signals[CHANGED], 0, namespace, key, value);
+}
+
+/**
+ * xdp_settings_read_value:
+ * @settings: the [class@Settings] object.
+ * @namespace: the namespace of the value.
+ * @key: the key of the value.
+ * @cancellable: a GCancellable or NULL.
+ * @error: return location for error or NULL.
+ *
+ * Read a setting value within @namespace, with @key.
+ *
+ * Returns: (nullable) (transfer full): the value, or %NULL if not
+ * found. If @error is not NULL, then the error is returned.
+ */
+GVariant *
+xdp_settings_read_value (XdpSettings *settings, const char *namespace, const char *key, GCancellable *cancellable, GError **error)
+{
+  g_autoptr(GVariant) ret = NULL;
+  g_autoptr(GVariant) inner = NULL;
+
+  ret = g_dbus_connection_call_sync (settings->portal->bus,
+                                     PORTAL_BUS_NAME,
+                                     PORTAL_OBJECT_PATH,
+                                     SETTINGS_INTERFACE,
+                                     "ReadOne",
+                                     g_variant_new ("(ss)", namespace, key),
+                                     NULL,
+                                     G_DBUS_CALL_FLAGS_NONE,
+                                     5000,
+                                     cancellable, error);
+  if (!ret) {
+    g_debug ("DBus error");
+    return NULL;
+  }
+
+  g_variant_get (ret, "(v)", &inner);
+
+  return g_steal_pointer (&inner);
+}
+
+/**
+ * xdp_settings_read_all:
+ * @settings: the [class@Settings] object.
+ * @namespaces: List of namespaces to filter results by, supports
+ * simple globbing explained below..
+ * @cancellable: a GCancellable or NULL.
+ * @error: return location for error or NULL.
+ *
+ * Read a setting value within @namespace, with @key.
+ *
+ * Returns: (nullable) (transfer full): the value, or %NULL if not
+ * found. If @error is not NULL, then the error is returned.
+ */
+GVariant *
+xdp_settings_read_all (XdpSettings *settings, const char *const *namespaces, GCancellable *cancellable, GError **error)
+{
+  g_autoptr(GVariant) ret = NULL;
+  g_autoptr(GVariant) inner = NULL;
+
+  ret = g_dbus_connection_call_sync (settings->portal->bus,
+                                     PORTAL_BUS_NAME,
+                                     PORTAL_OBJECT_PATH,
+                                     SETTINGS_INTERFACE,
+                                     "ReadAll",
+                                     g_variant_new ("(^as)", namespaces),
+                                     NULL,
+                                     G_DBUS_CALL_FLAGS_NONE,
+                                     5000,
+                                     cancellable, error);
+  if (!ret)
+    {
+      g_debug ("DBus error");
+      return NULL;
+    }
```

Comment:
```suggestion
  if (!ret)
    {
      g_debug ("DBus error");
      return NULL;
    }
```

This debug message isn't helpful. Either create an inner `GError` so we can always print a message or just remove it.

Reply:
removed.